src/3d/terrain/qgsterraindownloader.cpp

```diff
@@ -0,0 +1,169 @@
+/***************************************************************************
+  qgsterraindownloader.cpp
+  --------------------------------------
+  Date                 : March 2019
+  Copyright            : (C) 2019 by Martin Dobias
+  Email                : wonder dot sk at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsterraindownloader.h"
+
+#include "qgslogger.h"
+#include "qgsrasterlayer.h"
+
+#include "qgsgdalutils.h"
+
+
+QgsTerrainDownloader::QgsTerrainDownloader()
+{
+  QString uri = "type=xyz&url=http://s3.amazonaws.com/elevation-tiles-prod/terrarium/{z}/{x}/{y}.png&zmax=15&zmin=0";
```

Comment:
Any chance we could have that be a QgsSettings.value("3d/xyz","http...") defaulting to the aws url? That'd allows advanced users to tweak value without need to compile source.

Also, assuming there's an HTTPS variant of the url, please use that, otherwise you'll have QGIS users leaking out their areas of interests via plain text http requests :)

Reply:
Fixed http->https

Added custom URL + min/max zoom level in QgsTerrainDownloader. This is not exposed to GUI yet as it will need a bit of refactoring of QgsDemHeightMapGenerator to do it properly...